it/trace-context-leak/src/test/java/com/linecorp/armeria/common/TraceRequestContextLeakTest.java

```diff
@@ -0,0 +1,227 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.common.util.SafeCloseable;
+import com.linecorp.armeria.server.ServiceRequestContext;
+import com.linecorp.armeria.testing.junit5.common.EventLoopExtension;
+import com.linecorp.armeria.testing.junit5.common.EventLoopGroupExtension;
+
+import io.netty.channel.EventLoop;
+import io.netty.channel.EventLoopGroup;
+
+class TraceRequestContextLeakTest {
+
+    @RegisterExtension
+    static final EventLoopExtension eventLoopExtension = new EventLoopExtension();
+
+    @RegisterExtension
+    static final EventLoopGroupExtension eventLoopGroupExtension = new EventLoopGroupExtension(2);
+
+    @Test
+    void singleThreadContextNotLeak() throws InterruptedException {
+        final AtomicBoolean isThrown = new AtomicBoolean(false);
+        final CountDownLatch latch = new CountDownLatch(2);
+
+        final EventLoop executor =  eventLoopExtension.get();
+
+        executor.execute(() -> {
+            final ServiceRequestContext ctx = newCtx("/1");
+            try (SafeCloseable ignore = ctx.push()) {
+                //ignore
+            } catch (Exception ex) {
+                isThrown.set(true);
+            } finally {
+                latch.countDown();
+            }
+        });
+
+        executor.execute(() -> {
+            final ServiceRequestContext anotherCtx = newCtx("/2");
+            try (SafeCloseable ignore = anotherCtx.push()) {
+                //ignore
+            } catch (Exception ex) {
+                isThrown.set(true);
+            } finally {
+                latch.countDown();
+            }
+        });
+
+        latch.await();
+        assertThat(isThrown).isFalse();
```

Comment:
Simply use `await().untilTrue(isThrown)`

Reply:
For `Leak` test cases, using `await()` looks nicer, so I'll use it. But for `NotLeak` cases I think using `CountdownLatch` is suited more because I want to make sure that an exception isn't thrown and executors' tasks are finished. Thanks for the suggestion🙏.